lib/TH/generic/THVector.h

```diff
@@ -3,10 +3,12 @@
 #else
 
 TH_API void THVector_(fill)(real *x, const real c, const ptrdiff_t n);
+TH_API void THVector_(cadd)(real *z, const real *x, const real *y, const real c, const ptrdiff_t n);
 TH_API void THVector_(add)(real *y, const real *x, const real c, const ptrdiff_t n);
-TH_API void THVector_(diff)(real *z, const real *x, const real *y, const ptrdiff_t n);
-TH_API void THVector_(scale)(real *y, const real c, const ptrdiff_t n);
-TH_API void THVector_(mul)(real *y, const real *x, const ptrdiff_t n);
+TH_API void THVector_(cmul)(real *z, const real *x, const real *y, const ptrdiff_t n);
+TH_API void THVector_(mul)(real *y, const real *x, const real c, const ptrdiff_t n);
+TH_API void THVector_(cdiv)(real *z, const real *x, const real *y, const ptrdiff_t n);
+TH_API void THVector_(div)(real *y, const real *x, const real c, const ptrdiff_t n);
```

Comment:
Have you tried adding `THVector_(copy)` here? I know it sounds silly but I think AVX copy may be faster. And it's more common than these other ops.